ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1434,6 +1433,7 @@ void checkAndMaybeComplete() {
             logger.trace("{}: Attempt to put chunk with id: {} failed, attempting slipped put ", loggingContext,
                 chunkBlobId);
             routerMetrics.slippedPutAttemptCount.inc();
+            chunkException = null;
```

Comment:
Do we have issue here? Looks like either way, chunkException will be reset before the PutBlob retry, right?
Somehow for PutBlob, I like the current implementation more since we consolidate all the initial setting all in  prepareForSending.
But your call.

Reply:
right, it's just prepareForSending should not reset the chunkException. The exception should be reset in the checkAndMaybeComplete method. It's easier to understand that way.